### PR TITLE
[wasm] Avoid unnecessary relinking when publishing a blazor project for AOT

### PR DESCRIPTION
**Pull request by radical**
`Publish` target triggers `Build` target to run. When publishing, we don't want
to run relinking step during the `Build`, as it will be run for `Publish`
anyway. Earlier there wasn't a good way to differentiate the two cases of
`build` target - `dotnet build`, and `dotnet publish`, but now the sdk sets
`$(_IsPublishing)=true`, which can be used here.

- VS uses `$(DeployOnBuild)`, and that is still supported

- And add tests to run blazor+AOT

### CODE REVIEW

**Comment by radical** [comment]
Depends on https://github.com/dotnet/xharness/pull/1001

**Comment by radical** [comment]
This adds missing tests that run AOT'ed blazor app. And that repros the currently broken state.

**Review by maraf — APPROVED** [review]
Looks good to me 👍

File: src/mono/wasm/build/WasmApp.targets [hunk 1]
```diff
@@ -172,7 +173,7 @@
     <!-- Use a unique property, so the already run wasm targets can also run -->
     <MSBuild Projects="$(MSBuildProjectFile)"
```
Comment (pavelsavara) [inline src/mono/wasm/build/WasmApp.targets:174]: I'm curious how this works.
Do we run separate MSBuild process here (which doesn't share variables and items from parent ?)
Do we still need to have it as separate process ?
Do we still need `_WasmInNestedPublish_UniqueProperty_XYZ` ?

Comment (radical) [reply src/mono/wasm/build/WasmApp.targets:174]: > Do we run separate MSBuild process here (which doesn't share variables and items from parent ?)

No, but it runs it in a fresh context. I will be removing this, but I'm waiting for the blazor<->runtime build changes to happen.

> Do we still need to have it as separate process ?

It's not necessarily a separate process. And this PR doesn't change anything about this.

> Do we still need _WasmInNestedPublish_UniqueProperty_XYZ ?

Yep

Essentially, this PR doesn't change any behavior w.r.t the nested publish. The nested publish bit was a hacky solution, and will be removed. But I'm holding that off till we get further on the build consolidation with blazor.

Comment (pavelsavara) [reply src/mono/wasm/build/WasmApp.targets:174]: @ilonatommy opened issue for ICU related dependency on not nesting it 
https://github.com/dotnet/runtime/issues/82819

**Comment by radical** [comment]
Added the xharness dependency update here, from https://github.com/dotnet/runtime/pull/82600

**Comment by radical** [comment]
The failures are unrelated and https://github.com/dotnet/runtime/issues/82868 .

**Review by pavelsavara — APPROVED** [review]

**Pull request merged**